src/page-manipulation.js

```diff
@@ -26,149 +48,91 @@ function populateDailyData(card, repo, userId, date) {
-    return `<li class="historical-list-listItem">Your friend ${friendChallengeData} average steps.</li>`
-  }).join('');
+function populateWeeklyData(repo, userId) {
+  const calendar = document.querySelectorAll('.historic-data')
+  let date = document.querySelector('select').value
+  let week = repo.presentWeek(date, userId)
+  for(var i = 0; i < 8; i++) {
+    populateDailyData(calendar[i].id, repo, userId, week[i]);
+  }
 }
 
-function makeStepStreakHTML(id, activityInfo, userStorage, method) {
-  return method.map(streakData => {
-    return `<li class="historical-list-listItem">${streakData}!</li>`
-  }).join('');
+function displayWeeklyData(event, repo, id) {
+  let weeklyHtml = createWeeklyLayoutHtml()
+  insertWeekLayout(weeklyHtml[event.target.id])
+  populateWeeklyData(repo, id)
 }
 
+function populateUserInfo(user, userRepo) {
+  populateUserSidebar(user, userRepo);
+  populateUserCard(user, userRepo);
+  populateInfoCard(user);
+} 
 
-function addInfoToUserSidebar(user, userStorage) {
-  const sidebar = document.getElementById('user-sidebar');
-  const headerText = document.getElementById('headerText');
-  const accountInfo = document.getElementById('account-info');
-  const userGoals = document.getElementById('user-goals');
-  const accountInfoHtml = `
-    <p class="sidebar-friendslist-header" id="userAddress">${user.address}</p>
-    <div class="sidebar-header-line"></div>
-    <p class="sidebar-friendslist-header" id="userEmail">${user.email}</p>`
-
-  const userGoalsHtml = `
-    <p class="sidebar-friendslist-header" id="userStridelength">
-      Your stride length is ${user.strideLength} meters.
-    </p>
-    <div class="sidebar-header-line"></div>
-    <p class="sidebar-friendslist-header" id="stepGoalCard">
-      Your daily step goal is ${user.dailyStepGoal}
-    </p>
-    <p class="sidebar-friendslist-header" id="avStepGoalCard">
-      The average daily step goal is ${userStorage.calculateAverageStepGoal()}
-    <p>`
-
-  const leftSidebarHtmlBlock = `
-    <div class="sidebar-header-line"></div>
-    <section class="sidebar-body-friendContainer">
-      <p class="sidebar-friendslist-header">Friends</p>
-      <section class="sidebar-friendContainer-listItems">
-        <ul class="card-vertical-list" id="friendList">
-          ${makeFriendHTML(user, userStorage)}
-        </ul>
-      </section>
-      <div class="sidebar-header-line"></div>
-      <p class="this-week">
-        Keep up the good work! You were increasingly active on these dates:
-      </p>
-      <ul class="card-vertical-list" id="streakListMinutes">
-        <!-- friend list goes here -->
-      </ul>
-    </section>`;
-  headerText.insertAdjacentHTML('afterbegin', `${user.name}'s `);
-  accountInfo.insertAdjacentHTML('beforeend', accountInfoHtml);
-  sidebar.insertAdjacentHTML('beforeend', leftSidebarHtmlBlock);
-  userGoals.insertAdjacentHTML('beforeend', userGoalsHtml);
+function populateUserSidebar(user, repo) {
```

Comment:
much cleaner, than before looks awesome. I am probably gonna be stealing this for future projects.

Reply:
^Agreed for sure.